test/e2e/l7networkpolicy_test.go

```diff
@@ -216,7 +217,26 @@ func testL7NetworkPolicyHTTP(t *testing.T, data *TestData) {
 	require.NoError(t, NewPodBuilder(serverPodName, data.testNamespace, agnhostImage).OnNode(nodeName(0)).WithCommand(cmd).WithLabels(serverPodLabels).Create(data))
 	podIPs, err := data.podWaitForIPs(defaultTimeout, serverPodName, data.testNamespace)
 	require.NoError(t, err, "Expected IP for Pod '%s'", serverPodName)
-	serverIPs := podIPs.AsSlice()
+	dstPodIPs := podIPs.AsSlice()
+
+	// Create a Service whose backend is the above backend Pod.
+	var ipFamilies []corev1.IPFamily
+	if len(clusterInfo.podV4NetworkCIDR) != 0 {
+		ipFamilies = append(ipFamilies, corev1.IPv4Protocol)
+	}
+	if len(clusterInfo.podV6NetworkCIDR) != 0 {
+		ipFamilies = append(ipFamilies, corev1.IPv6Protocol)
+	}
+	mutator := func(service *corev1.Service) {
+		service.Spec.IPFamilies = ipFamilies
+	}
+	svc, err := data.CreateServiceWithAnnotations("svc-agnhost", data.testNamespace, p8080, p8080, corev1.ProtocolTCP, serverPodLabels, false, false, corev1.ServiceTypeClusterIP, nil, nil, mutator)
+	require.NoError(t, err)
+	var serviceIPs []*net.IP
+	for _, clusterIP := range svc.Spec.ClusterIPs {
```

Comment:
Is it safe to assume that the ClusterIPs are ready immediately after the Service is created?

Reply:
That's a good question. It seems that we have already made the assumption that the the ClusterIPs are available in almost every tests using the function `CreateServiceWithAnnotations`.

Reply:
I may be wrong, but I think it's guaranteed to be the case. The Service object returned by `Create` should have the field set.

Reply:
clusterIP and nodePort are allocated synchronously by kube-apiserver